### PR TITLE
MM-34675 Improve UI for system console toggle.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Improved UI for system console toggle. Went ahead and used the updated UI instead of the one specified in the ticket. 

New UI:
image

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34675

### CODE REVIEW

File: webapp/src/system_console_enabled_teams.tsx [hunk 1]
```diff
@@ -53,18 +54,53 @@ const TeamSelector: FC<TeamSelectorProps> = (props: TeamSelectorProps) => {
         }
     };
 
+    const radioPressed = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e.target.value === 'enabled') {
+            setEnabled(true);
+        } else {
+            props.onTeamsSelected([]);
```
Comment (lieut-data) [inline webapp/src/system_console_enabled_teams.tsx:61]: Does this result in forcibly clearing the selected teams? Seems slightly counterintuitive for how I expect dependent fields to behave: i.e. if I disable LDAP, I don't lose all my LDAP configuration.

File: webapp/src/system_console_enabled_teams.tsx [hunk 2]
```diff
@@ -53,18 +54,53 @@ const TeamSelector: FC<TeamSelectorProps> = (props: TeamSelectorProps) => {
...
+                </RadioLabel>
+                <RadioLabel>
+                    <RadioInput
+                        type='radio'
+                        name='enabled'
+                        value='enabled'
+                        checked={enabled}
+                        onChange={radioPressed}
+                    />
+                    {'Incident Collaboration enabled on slected teams.'}
```
Comment (lieut-data) [inline webapp/src/system_console_enabled_teams.tsx:87]:
```suggestion
                    {'Incident Collaboration enabled on selected teams.'}
```

**Review by lieut-data — APPROVED** [review]
A few comments, but don't need to block on a re-review from me.

**Comment by lieut-data** [comment]
@agarciamontoro, we're considering this for cherry-picking to 1.9 -- can you prioritize review?

File: webapp/src/system_console_enabled_teams.tsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ type GetTeamType = (teamID: string) => Team
 const TeamSelector: FC<TeamSelectorProps> = (props: TeamSelectorProps) => {
     const selectableTeams = useSelector<GlobalState, Team[]>(getMyTeams);
     const getTeamFromID = useSelector<GlobalState, GetTeamType>((state) => (teamId) => getTeam(state, teamId) || {display_name: 'Unknown Team', id: teamId});
+    const [enabled, setEnabled] = useState(Boolean(props.teamsSlected) && props.teamsSlected.length !== 0);
```
Comment (lieut-data) [inline webapp/src/system_console_enabled_teams.tsx:23]: Typo here as well: "teamSelected".

**Comment by itao** [comment]
Can we change the toggle text to: `Enable for every team` and `Enable for selected teams only`. Repeating Incident Collaboration here feels redundant

**Review by agarciamontoro — APPROVED** [review]
Apart from Jesse's comments, it looks good!

File: webapp/src/system_console_enabled_teams.tsx (hunk 1 above)
Comment (crspeller) [reply webapp/src/system_console_enabled_teams.tsx:61]: Yup. Given the UX, I don't think it's worth the complexity to save the list of teams selected even when you disable it. I don't really see a use case for keeping the list of teams around.

**Review by lieut-data — APPROVED** [review]

**Comment by lieut-data** [comment]
Huh, I didn't even realize `mattermod` would try in this repo! Guess I should have setup the release branch in advance :p

**Comment by crspeller** [comment]
Nice try @mattermod 😆

**Pull request merged**